Mage.Sets/src/mage/cards/c/ChainsOfCustody.java

```diff
@@ -0,0 +1,66 @@
+package mage.cards.c;
+
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.GenericManaCost;
+import mage.abilities.effects.common.AttachEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.keyword.EnchantAbility;
+import mage.abilities.keyword.WardAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.AttachmentType;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.filter.StaticFilters;
+import mage.target.TargetPermanent;
+import mage.abilities.common.EntersBattlefieldTriggeredAbility;
+import mage.abilities.common.delayed.OnLeaveReturnExiledToBattlefieldAbility;
+import java.util.UUID;
+import mage.abilities.effects.common.CreateDelayedTriggeredAbilityEffect;
+import mage.abilities.effects.common.ExileUntilSourceLeavesEffect;
+/**
+
+ * @author Chesse20
+ */
+public final class ChainsOfCustody extends CardImpl {
+
+
+    public ChainsOfCustody(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{2}{W}");
+
+        this.subtype.add(SubType.AURA);
+
+        // Enchant creature *YOU* control
+        TargetPermanent auraTarget = new TargetPermanent(StaticFilters.FILTER_CONTROLLED_CREATURE); //only your creatures!
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.BoostCreature));
+        this.addAbility(new EnchantAbility(auraTarget));
+
+        //exile until leaves the battlefield
+        Ability ability = new EnchantAbility(auraTarget);
+        ability = new EntersBattlefieldTriggeredAbility(new ExileUntilSourceLeavesEffect());
+        ability.addTarget(new TargetPermanent(StaticFilters.FILTER_OPPONENTS_PERMANENT_NON_LAND)); //more than just creatures
+        ability.addEffect(new CreateDelayedTriggeredAbilityEffect(new OnLeaveReturnExiledToBattlefieldAbility()));
+        this.addAbility(ability);
+
+        // ward 2
+        this.addAbility(new SimpleStaticAbility(
+                new GainAbilityAttachedEffect(
+                        new WardAbility(new GenericManaCost(2), false), AttachmentType.AURA
+                )
```

Comment:
Please apply `.setText("Enchanted creature has ward {2}.")` to `GainAbilityAttachedEffect` for desired rule text

Reply:
should be fixed